### PR TITLE
Use package:oidc

### PR DESCRIPTION
**Pull request by ahmednfwela**
This PR changes the repository to use https://pub.dev/packages/oidc

This PR also makes running with vercel smoother, by delegating the zitadel configuration part (url + clientId) to build time instead of making the user change it manually.

so to build the flutter app, now you do:
```bash
flutter build web --dart-define zitadel_url=[zitadel-url] --dart-define zitadel_client_id=[zitadel-client-id]
```
and to debug it:
```bash
flutter run -d chrome --web-port=4444 --dart-define zitadel_url=[zitadel-url] --dart-define zitadel_client_id=[zitadel-client-id]
```

Another PR will be created to update the flutter guide [here](https://github.com/zitadel/zitadel/blob/main/docs/docs/examples/login/flutter.md)

### CODE REVIEW

**Comment by ahmednfwela** [comment]
cc @peintnermax

**Comment by peintnermax** [comment]
@ahmednfwela thx for your PR. We're currently very busy and will address this once we have resolved our most critical stuff 👍

File: README.md [hunk 1]
```diff
@@ -24,61 +38,38 @@ install command is
...
-        <action android:name="android.intent.action.VIEW" />
-        <category android:name="android.intent.category.DEFAULT" />
-        <category android:name="android.intent.category.BROWSABLE" />
-        <data android:scheme="[callback-url-scheme]" /> <!-- ex: com.example.zitadelflutter -->
-    </intent-filter>
-</activity>
-```
-
-then connect your device or run a simulator and run your application.
+1. Copy your instance url and your `clientId` and set it in `lib/main.dart` to the `zitadelIssuer` and `zitadelClientId` variables.
```
Comment (peintnermax) [inline README.md:53]: This is not required anymore right?

**Review by peintnermax — COMMENTED** [review]
running the application on the iphone simulator gets me an error.. You know what could be the problem?
```
Exception: Binding has not yet been initialized.
The "instance" getter on the ServicesBinding binding mixin is only available once that binding has been initialized.
Typically, this is done by calling "WidgetsFlutterBinding.ensureInitialized()" or "runApp()" (the latter calls the former). Typically this call is done in the "void main()" method. The "ensureInitialized" method is idempotent; calling it multiple times is not harmful. After calling that method, the "instance" getter will return the binding.
In a test, one can call "TestWidgetsFlutterBinding.ensureInitialized()" as the first line in the test's "main()" method to initialize the binding.
If ServicesBinding is a custom binding mixin, there must also be a custom binding class, like WidgetsFlutterBinding, but that mixes in the selected binding, and that is the class that must be constructed before using the "instance" getter.
```

File: README.md (hunk 1 above)
Comment (peintnermax) [reply README.md:53]: Could you elaborate a little more on this error? Thx

**Comment by ahmednfwela** [comment]
oh this should be a simple fix that gets added to the top of the main method

File: README.md (hunk 1 above)
Comment (ahmednfwela) [reply README.md:53]: it's not required, but it's simpler to copy the variables than having to pass the environment variable each build

**Comment by peintnermax** [comment]
The ios app is working fine now. Thx for your contribution.

**Review by peintnermax — APPROVED** [review]

**Pull request merged**